# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/vr/ipc/VRLayerChild.h
# Commit: dd00e2da3a0f
# Full Hash: dd00e2da3a0f11a7cc0af9e5ece21a339f4393eb
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 69cc1de6f754 -r dd00e2da3a0f gfx/vr/ipc/VRLayerChild.h
--- a/gfx/vr/ipc/VRLayerChild.h	Wed Jun 10 21:57:19 2020 +0000
+++ b/gfx/vr/ipc/VRLayerChild.h	Wed Jun 10 22:21:02 2020 +0000
@@ -50,7 +50,7 @@
   virtual void ActorDestroy(ActorDestroyReason aWhy) override;
 
   RefPtr<dom::HTMLCanvasElement> mCanvasElement;
-  bool mIPCOpen;
+  bool mIPCOpen = false;
 
   // AddIPDLReference and ReleaseIPDLReference are only to be called by
   // CreateIPDLActor and DestroyIPDLActor, respectively. We intentionally make
@@ -64,10 +64,10 @@
   gfx::Rect mRightEyeRect;
   RefPtr<WebGLFramebufferJS> mFramebuffer;
 
-  RefPtr<layers::SharedSurfaceTextureClient> mThisFrameTexture;
-  RefPtr<layers::SharedSurfaceTextureClient> mLastFrameTexture;
+  Maybe<layers::SurfaceDescriptor> mThisFrameTextureDesc;
+  Maybe<layers::SurfaceDescriptor> mLastFrameTextureDesc;
 
-  uint64_t mLastSubmittedFrameId;
+  uint64_t mLastSubmittedFrameId = 0;
 };
 
 }  // namespace gfx